File: frontend/lib/data/PostData.dart
import 'package:json_annotation/json_annotation.dart';
import 'package:size_hub/data/UserMeasurementsData.dart';
import 'package:size_hub/data/database/Database.dart';

part 'PostData.g.dart';

@JsonSerializable()
class PostData {
  final String id;
  final String userId;
  final String clothingItemId;
  final List<String> photoUrls;
  final UserMeasurementsData userMeasurementsData;

  PostData(this.id, this.userId, this.clothingItemId, this.photoUrls,
      this.userMeasurementsData);

  factory PostData.fromJson(Map<String, dynamic> json) =>
      json == null ? null : _$PostDataFromJson(json);

  Map<String, dynamic> toJson() => _$PostDataToJson(this);

  Future<String> getUsername(Database database) async {
    return await database.getUsername(userId);
  }

  @override
  String toString() {
    return 'PostData{id: $id, userId: $userId, photoUrls: $photoUrls, clothingItemId: $clothingItemId, userMeasurementData: $userMeasurementsData}';
  }
}

File: backend/src/main/kotlin/data/PostData.kt
package src.data

import kotlinx.serialization.Serializable

@Serializable
class PostData(val userId: Id,
               val clothingItemId: Id? = null,
               val clothingItemSize: String? = null,
               val userMeasurementsData: UserMeasurementsData? = null,
               val photoUrls: List<String>,
               val location: PostLocation? = null,
               val caption: String,
               val createdAt: String)